scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,553 @@
+            default=None,
+            nargs="?",
+            metavar="SDK_VER",
+            help="specifies the install version of Zephyr SDK. "
+            "If not specified, the install version detects from "
+            "${ZEPHYR_BASE}/SDK_VERSION file.",
+        )
+        install_args_parser.add_argument(
+            "-d",
+            "--install-dir",
+            default=None,
+            metavar="DIR",
+            help="specifies the install destination directory.",
+        )
+        install_args_parser.add_argument(
+            "-i",
+            "--interactive",
+            action="store_true",
+            help="launches installer in interactive mode. "
+            "--toolchains, --no-toolchains and --no-hosttools will be ignored if this option enabled.",
+        )
+        install_args_parser.add_argument(
+            "-t",
+            "--toolchains",
+            metavar="TC",
+            nargs="+",
+            help="specifies the installation of toolchains. "
+            "If this option is not given, all toolchains will be installed.",
+        )
+        install_args_parser.add_argument(
+            "-T",
+            "--no-toolchains",
+            action="store_true",
+            help="do not install toolchains. "
+            "--toolchains will be ignored if this option is enabled.",
+        )
+        install_args_parser.add_argument(
+            "-H",
+            "--no-hosttools",
+            action="store_true",
+            help="do not install host-tools.",
+        )
+        install_args_parser.add_argument(
+            "--personal-access-token", help="specify GitHub personal access token."
+        )
+        install_args_parser.add_argument(
+            "--api-url",
+            default="https://api.github.com/repos/zephyrproject-rtos/sdk-ng/releases",
+            help="specifies the GitHub releases API endpoint for finding SDKs.",
+        )
+
+        return parser
+
+    def os_arch_name(self):
+        system = platform.system()
+        machine = platform.machine()
+
+        if system == "Linux":
+            osname = "linux"
+        elif system == "Darwin":
+            osname = "macos"
+        elif system == "Windows":
+            osname = "windows"
+        else:
+            self.die(f"Unsupported system: {system}")
+
+        if machine in ["aarch64", "arm64"]:
+            arch = "aarch64"
+        elif machine in ["x86_64", "AMD64"]:
+            arch = "x86_64"
+        else:
+            self.die(f"Unsupported machine: {machine}")
+
+        return (osname, arch)
+
+    def detect_version(self, args):
+        if args.version:
+            version = args.version
+        else:
+            if os.environ["ZEPHYR_BASE"]:
+                zephyr_base = os.environ["ZEPHYR_BASE"]
+            else:
+                zephyr_base = os.path.join(os.path.dirname(__file__), "..", "..")
+
+            if not os.path.exists(os.path.join(zephyr_base, "SDK_VERSION")):
```

Comment:
I would recommend switching to pathlib across the entire file

```suggestion
            if not (Path(zephyr_base) / "SDK_VERSION").exists():
```

Reply:
Fixed it. 
I revisited this point overall.